src/lib/logical_query_plan/lqp_translator.cpp

```diff
@@ -293,26 +293,36 @@ std::shared_ptr<AbstractOperator> LQPTranslator::_translate_join_node(
     return std::make_shared<Product>(input_left_operator, input_right_operator);
   }
 
-  Assert(join_node->join_predicate(), "Need predicate for non Cross Join");
-
-  /**
-   * Assert that the Join Predicate is simple, e.g. of the form <column_a> <predicate> <column_b>.
-   * We do not require <column_a> to be in the left input though.
-   */
-  const auto operator_join_predicate =
-      OperatorJoinPredicate::from_expression(*join_node->join_predicate(), *node->left_input(), *node->right_input());
-  Assert(operator_join_predicate,
-         "Couldn't translate join predicate: "s + join_node->join_predicate()->as_column_name());
+  Assert(!join_node->join_predicates().empty(), "Need predicate for non Cross Join");
+
+  std::optional<OperatorJoinPredicate> primary_operator_join_predicate;
+  std::vector<OperatorJoinPredicate> secondary_operator_join_predicates{};
+
+  for (const auto& predicate_expression : join_node->join_predicates()) {
+    auto operator_join_predicate =
+        OperatorJoinPredicate::from_expression(*predicate_expression, *node->left_input(), *node->right_input());
+    // Assert that the Join Predicates are simple, e.g. of the form <column_a> <predicate> <column_b>.
+    // We do not require <column_a> to be in the left input though.
+    Assert(operator_join_predicate, "Couldn't translate join predicate: "s + predicate_expression->as_column_name());
+    if (!primary_operator_join_predicate) {
+      primary_operator_join_predicate = operator_join_predicate.value();
+    } else {
+      secondary_operator_join_predicates.emplace_back(operator_join_predicate.value());
```

Comment:
This looks like storing all in the same vector (and maybe extracting the primary predicate into its own reference later) could be easier